### PR TITLE
don't wait forever while waiting for an inv-sync pass

### PR DESCRIPTION
**Pull request by jcnelson**
Fixes https://discord.com/channels/621759717756370964/730814571517968436/829186890388340816

### CODE REVIEW

File: testnet/stacks-node/src/syncctl.rs [hunk 1]
```diff
@@ -62,28 +62,17 @@ impl PoxSyncWatchdogComms {
...
-        }
-        return true;
-    }
-
     /// Wait for at least one inv-sync state-machine passes
     pub fn wait_for_inv_sync_pass(&self, timeout: u64) -> Result<bool, burnchain_error> {
         let current = self.get_inv_sync_passes();
 
+        let now = get_epoch_time_secs();
         while current >= self.get_inv_sync_passes() {
```
Comment (jcnelson) [inline testnet/stacks-node/src/syncctl.rs:70]: TL;DR We have to keep polling `self.get_inv_sync_passes()` or this loop will never exit.

**Review by lgalabru — APPROVED** [review]
LGTM, thank you @jcnelson!

**Review by kantai — APPROVED** [review]
LGTM!

**Comment by diwakergupta** [comment]
Thanks @jcnelson ! Is it possible to add an integration or E2E test that would catch a regression going forward? The fact we had built and deployed an RC2 without catching this type of "dead on arrival" behavior isn't great.

**Pull request merged**